ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/config/AuditYamlConfigurationLoader.java

```diff
@@ -90,53 +110,52 @@ public AuditConfig loadConfig() throws ConfigurationException
         }
     }
 
-    private static URL getConfigURL(Properties properties)
+    private URL getConfigURL()
     {
-        String propertiesPath = properties.getProperty(PROPERTY_CONFIG_FILE, DEFAULT_CONFIG_FILE);
-        if (!propertiesPath.isEmpty())
+        String customConfigPath = properties.getProperty(PROPERTY_CONFIG_FILE);
+
+        URL url;
+        if (customConfigPath != null)
         {
+            LOG.debug("Looking for audit config at " + customConfigPath);
             try
             {
-                URL url = new File(propertiesPath).toURI().toURL();
+                url = new File(customConfigPath).toURI().toURL();
                 url.openStream().close();
-                return url;
             }
             catch (IOException e)
             {
-                throw new ConfigurationException("Failed to load configuration from " + propertiesPath, e);
+                throw new ConfigurationException("Failed to load configuration from " + customConfigPath, e);
             }
         }
+        else
```

Comment:
The code here is a bit confusing (even more so when DEFAULT_CONFIG_FILE is considered).
loadConfig() / getConfigURL()
Both methods open a stream towards the config file but here we only throw exception if the PROPERTY_CONFIG_FILE could not be opened...
Maybe this method could return a Stream (or null) instead? Then the file "finding/fiddling" could be done at one place
Also there would be no need for a comment like this - "// Shouldn't happen if getConfigURL() is used" :-)

Reply:
Yeah, this is messy for sure. I was trying to keep changes to a minimum, but perhaps I should break this down a bit. Right now it is kind of hard to tell if all strange things are intentional or not.